CMakeLists.txt

```diff
@@ -0,0 +1,68 @@
+cmake_minimum_required(VERSION 3.9)
+
+# Fail immediately if not using an out-of-source build
+if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_CURRENT_BINARY_DIR)
+  message(FATAL_ERROR
+    "In-source builds are not supported.  Please create a build directory "
+    "separate from the source directory")
+endif ()
+
+#------------------------------------------------------------------------------#
+# Parse version number from H5Zzfp_plugin.h
+#------------------------------------------------------------------------------#
+file(STRINGS "${CMAKE_CURRENT_SOURCE_DIR}/src/H5Zzfp_plugin.h" H5Z_ZFP_H REGEX "^\#define H5Z_FILTER_ZFP_VERSION_MAJOR")
+string(REGEX REPLACE "^.*MAJOR " "" H5Z_ZFP_VERSION_MAJOR "${H5Z_ZFP_H}")
+file(STRINGS "${CMAKE_CURRENT_SOURCE_DIR}/src/H5Zzfp_plugin.h" H5Z_ZFP_H REGEX "^\#define H5Z_FILTER_ZFP_VERSION_MINOR")
+string(REGEX REPLACE "^.*MINOR " "" H5Z_ZFP_VERSION_MINOR "${H5Z_ZFP_H}")
+file(STRINGS "${CMAKE_CURRENT_SOURCE_DIR}/src/H5Zzfp_plugin.h" H5Z_ZFP_H REGEX "^\#define H5Z_FILTER_ZFP_VERSION_PATCH")
+string(REGEX REPLACE "^.*PATCH " "" H5Z_ZFP_VERSION_PATCH "${H5Z_ZFP_H}")
+set(H5Z_ZFP_VERSION "${H5Z_ZFP_VERSION_MAJOR}.${H5Z_ZFP_VERSION_MINOR}.${H5Z_ZFP_VERSION_PATCH}")
+
+project(H5Z_ZFP VERSION ${H5Z_ZFP_VERSION} LANGUAGES C)
+
+#------------------------------------------------------------------------------#
+# Compile options.
+#------------------------------------------------------------------------------#
+option(FORTRAN_INTERFACE "Enable the Fortran interface" ON)
+if (FORTRAN_INTERFACE)
+  enable_language(Fortran)
+endif ()
+
+#------------------------------------------------------------------------------#
+# Some boilerplate to setup nice output directories
+#------------------------------------------------------------------------------#
+set(CMAKE_INSTALL_INCLUDEDIR include)
+set(CMAKE_INSTALL_CMAKEDIR lib/cmake/h5z_zfp)
+
+#------------------------------------------------------------------------------#
+# Required packages
+#------------------------------------------------------------------------------#
+# Find HDF5, relies on HDF5_ROOT being set in environment.
```

Comment:
This note mentions need to define `HDF5_ROOT` in environment. Is it also possible to do `cmake -DHDF5_ROOT:PATH=/path/to/hdf5` or is there a slightly differently named variable for same when being passed on command-line to CMake?

Reply:
Interesting question. Yes, you can by using `CMAKE_PREFIX_PATH` variable on the command-line. I have a short explanation in the `cmake.rst` how is done.